server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -237,7 +237,9 @@ public Storage selectStorage(ShuffleDataReadEvent event) {
   @Override
   public void updateWriteMetrics(ShuffleDataFlushEvent event, long writeTime) {
     super.updateWriteMetrics(event, writeTime);
-    ShuffleServerMetrics.counterTotalLocalFileWriteDataSize.inc(event.getSize());
+    ShuffleServerMetrics.counterTotalLocalFileWriteDataSize
+        .labels(event.getUnderStorage().getStoragePath())
```

Comment:
I think we should check wheater the underStorage is null?

Reply:
done.